src/app/store/assets/utils.ts

```diff
@@ -1,17 +1,28 @@
-import { fetchTokensMetadata, isKnownChainId, TokenMetadataResponse } from 'lib/apis/temple';
-import { fetchTzktAccountAssets, fetchTzktAccountAssetsPage } from 'lib/apis/tzkt';
-import { TzktAccountAsset } from 'lib/apis/tzkt/types';
+import pMemoize from 'p-memoize';
+
+import { fetchTokensMetadata, isKnownChainId } from 'lib/apis/temple';
+import { fetchTzktAccountAssets } from 'lib/apis/tzkt';
+import type { TzktAccountAsset } from 'lib/apis/tzkt/types';
 import { toTokenSlug } from 'lib/assets';
 import { isCollectible } from 'lib/metadata';
+import type { FetchedMetadataRecord } from 'lib/metadata/fetch';
+
+import { MetadataMap } from '../collectibles-metadata/state';
+
+export const getAccountAssetsStoreKey = (account: string, chainId: string) => `${account}@${chainId}`;
 
-import type { MetadataRecords } from '../tokens-metadata/state';
+export const isAccountAssetsStoreKeyOfSameChainIdAndDifferentAccount = (
+  key: string,
+  account: string,
+  chainId: string
+) => !key.startsWith(account) || key.endsWith(chainId);
```

Comment:
Sorry, something mismatches here. Did you mean `isAccountAssetsStoreKeyOfSameChainIdOrDifferentAccount` or `!key.startsWith(account) && key.endsWith(chainId)`?